homeassistant/components/datadog/config_flow.py

```diff
@@ -0,0 +1,85 @@
+"""Config flow for Datadog."""
+
+from typing import Any
+
+from datadog import DogStatsd
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry, ConfigFlowResult, OptionsFlow
+from homeassistant.core import callback
+
+from . import validate_datadog_connection
+from .const import DEFAULT_HOST, DEFAULT_PORT, DEFAULT_PREFIX, DEFAULT_RATE, DOMAIN
+
+
+class DatadogConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Datadog."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle user config flow."""
+        errors: dict[str, str] = {}
+        if user_input:
+            # Validate connection to Datadog Agent
+            success = await validate_datadog_connection(
+                DogStatsd(user_input["host"], user_input["port"], user_input["prefix"])
+            )
+            if not success:
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_create_entry(title="Datadog", data=user_input)
```

Comment:
```suggestion
                return self.async_create_entry(title="Datadog", data={}, options=user_input)
```
As all these can change, might as well store them in `options` directly and not use `data`